modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/api/construction/ConstructionPayloads.scala

```diff
@@ -0,0 +1,44 @@
+package org.tessellation.rosetta.domain.api.construction
+
+import cats.data.NonEmptyList
+
+import org.tessellation.ext.derevo.magnoliaCustomizable.snakeCaseConfiguration
+import org.tessellation.rosetta.domain.api.construction.ConstructionMetadata.TransactionMetadata
+import org.tessellation.rosetta.domain.operation.Operation
+import org.tessellation.rosetta.domain.{NetworkIdentifier, RosettaPublicKey, SigningPayload}
+import org.tessellation.security.hex.Hex
+
+import derevo.cats.{eqv, show}
+import derevo.circe.magnolia.{customizableDecoder, customizableEncoder}
+import derevo.derive
+
+case object ConstructionPayloads {
+
+  @derive(customizableDecoder)
+  case class Request(
+    networkIdentifier: NetworkIdentifier,
+    operations: NonEmptyList[Operation],
+    metadata: Option[TransactionMetadata],
+    publicKeys: Option[NonEmptyList[RosettaPublicKey]]
```

Comment:
if we require public keys (what we actually do, because later on you return an error if there are missing public keys), it should be required field too. you should define your domain in a way to make impossible states not representable (as long as it's possible and don't require ton of overhead).

Reply:
edit: I probably mixed public keys with account identifiers, because you check acc. identifiers, not the public keys. but then, are public keys property used somehow? if we don't use it at all, then it should be removed

Reply:
>>> rosetta API specification is written to cover all possibilities, while we are writing an implementation for specific network, not generic one. the API should be crafted for our network then. by doing that, you would automatically not allow to accept a request that is invalid (because for example has no metadata, so you won't be able to construct a transaction).

If something is required, then we have to specify it as well. If something is optional, most likely rosetta specifies why it's optional and when it's required, when not.
Also, take a look there: https://www.rosetta-api.org/docs/common_mistakes.html#return-empty-value-or-null-for-optional-fields
It says that all properties that are null or optional (and hence null) should not be returned at all. If we don't use an optional property (so we will always return None), we can simply remove them. For requests - if there is an optional field, but our implementation requires that field to work correctly, we should specify it as required in our internal implementation (to fail-fast)